runtime/src/migration.rs

```diff
@@ -138,5 +251,53 @@ mod nomination_pools {
 
 			<Runtime as frame_system::Config>::DbWeight::get().writes(5u64)
 		}
+
+		/// Wrapper for all migrations of this pallet.
+		pub(crate) fn migrate() -> Weight {
+			let onchain_version = Pallet::<Runtime>::on_chain_storage_version();
+			let mut weight: Weight = Weight::zero();
+
+			if onchain_version < 1 {
+				weight = weight.saturating_add(v0_to_v1::on_runtime_upgrade());
+			}
+
+			v0_to_v1::STORAGE_VERSION.put::<Pallet<Runtime>>();
+			weight.saturating_add(<Runtime as frame_system::Config>::DbWeight::get().writes(1))
+		}
+	}
+
+	// Migrations that set `StorageVersion`s which was missed to set.
+	pub mod v1_to_v3 {
+		use super::*;
+
+		#[cfg(feature = "try-runtime")]
+		pub fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+			let storage_version = NominationPools::on_chain_storage_version();
+			log::info!(
+				"Old nomination pools storage version: {:?}",
+				storage_version
+			);
+			Ok(sp_std::vec![])
+		}
+
+		// Here we update the storage version to be consistent even if pallet was initialized correctly
+		pub fn on_runtime_upgrade() -> Weight {
+			let storage_version = NominationPools::on_chain_storage_version();
+			if storage_version < 3 {
+				StorageVersion::new(3).put::<NominationPools>();
```

Comment:
Essentially, for the Kate testnet, do we only need to increase the version to 3 without performing any actual data migration?

Reply:
yes, the data is already in the correct format, it's just a storage version mismatch after all. THe v3 to v5 will be done after storage version update